File: apps/study/e-minimum-diff-high-low/mdhl.go
package numsa

import (
	"math"
	"sort"
)

func Numsubarrays(nums []int, k int) int {
	sort.Ints(nums)

	l, r := 0, k-1
	res := math.MaxInt32

	for r < len(nums) {
		res = min(res, nums[r]-nums[l])
		l++
		r++
	}

	return res
}

File: apps/study/e-minimum-diff-high-low/mdhl.ts
export function minimumDiffHighLow(nums: number[], k: number) {
  nums.sort((a, b) => a - b)
  let l = 0
  let r = k - 1
  let res = Infinity

  while (r < nums.length) {
    res = Math.min(res, nums[r] - nums[l])
    l++
    r++
  }

  return res
}
